### PR TITLE
Replace DotNetty packages

### PR DESCRIPTION
**Pull request by rubo**
## Changes
Replaced `DotNetty` package with `Nethermind.DotNetty`.

## Types of changes

What types of changes does your code introduce?

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

## Further comments (optional)

Currently, the packages are hosted in a testing environment. If they prove to be good, we'll publish them on nuget.org.

### CODE REVIEW

File: src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj [hunk 1]
```diff
@@ -8,8 +8,8 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="DotNetty.Transport" Version="0.7.2" />
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.2.0" />
+        <PackageReference Include="Nethermind.DotNetty.Transport" Version="0.7.5-preview.1" />
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj:12]: why preview?

File: src/Nethermind/NuGet.Config [hunk 2]
```diff
@@ -1,6 +1,6 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+    <add key="int.nugettest.org" value="https://apiint.nugettest.org/v3/index.json" />
```
Comment (LukaszRozmej) [inline src/Nethermind/NuGet.Config:4]: shouldn't we have both?

File: src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj (hunk 1 above)
Comment (rubo) [reply src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj:12]: Cause we're testing it.

File: src/Nethermind/NuGet.Config (hunk 2 above)
Comment (rubo) [reply src/Nethermind/NuGet.Config:4]: We should not.

File: src/Nethermind/Nethermind.Network.Enr/Nethermind.Network.Enr.csproj [hunk 3]
```diff
@@ -8,12 +8,12 @@
...
 
     <ItemGroup>
-      <ProjectReference Include="..\Nethermind.Crypto\Nethermind.Crypto.csproj" />
-      <ProjectReference Include="..\Nethermind.Network\Nethermind.Network.csproj" />
+      <PackageReference Include="Nethermind.DotNetty.Buffers" Version="0.7.5-preview.1" />
     </ItemGroup>
 
     <ItemGroup>
-      <PackageReference Include="DotNetty.Buffers" Version="0.7.2" />
+      <ProjectReference Include="..\Nethermind.Crypto\Nethermind.Crypto.csproj" />
```
Comment (tanishqjasoria) [inline src/Nethermind/Nethermind.Network.Enr/Nethermind.Network.Enr.csproj:15]: is this needed here?

**Review by tanishqjasoria — APPROVED** [review]

File: src/Nethermind/Nethermind.Network.Enr/Nethermind.Network.Enr.csproj (hunk 3 above)
Comment (rubo) [reply src/Nethermind/Nethermind.Network.Enr/Nethermind.Network.Enr.csproj:15]: I don't know. I just replaced the package. Please check, if we don't, I'll remove it.

File: src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj:12]: Don't merge it to master then - this should be a draft

**Comment by LukaszRozmej** [comment]
@tanishqjasoria @rubo  Can you link changes to the package here?

File: src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj (hunk 1 above)
Comment (rubo) [reply src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj:12]: Just to clarify. I'm done with this PR and marked it as ready. It's now up to you what to do next, keep it a draft, merge, etc.

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj:12]: Lets run smoke tests + local nodes and if everything is fine mark it as stable and update this PR?

Comment (rubo) [reply src/Nethermind/Nethermind.Network.Discovery.Test/Nethermind.Network.Discovery.Test.csproj:12]: Alright

**Comment by LukaszRozmej** [comment]
@rubo lets make the package stable and make it mergable.

**Pull request merged**